packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -501,6 +479,52 @@ export class TokenDetectionController extends StaticIntervalPollingController<To
     return isEqualValues;
   }
 
+  #getCorrectNetworkClientIdByChainId(
+    chainIds: Hex[] | undefined,
+  ): { chainId: Hex; networkClientId: NetworkClientId }[] | null {
+    const { networkConfigurationsByChainId, selectedNetworkClientId } =
+      this.messagingSystem.call('NetworkController:getState');
+
+    if (!chainIds) {
```

Comment:
What's the case where chain ids is not defined?  Would it not always be passed?

Reply:
this is because the chainIds property is optional in the detectTokens function, so we default to the selected network when it's not provided.